src/main/java/htsjdk/samtools/read/Read.java

```diff
@@ -0,0 +1,1438 @@
+     *
+     * @param duplicate {@code true} if the read is a duplicate; {@code false} otherwise.
+     */
+    default void setDuplicate(final boolean duplicate) {
+        addFlag(SAMFlag.DUPLICATE_READ, duplicate);
+    }
+
+    /**
+     * Checks if the read is a supplementary alignment.
+     *
+     * <p>Equivalent to the 0x800 <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation calls {@link #isSet(SAMFlag)} for {@link SAMFlag#SUPPLEMENTARY_ALIGNMENT}.
+     *
+     * @return {@code true} if the read is a supplementary alignment; {@code false} otherwise.
+     */
+    default boolean isSupplementaryAlignment() {
+        return isSet(SAMFlag.SUPPLEMENTARY_ALIGNMENT);
+    }
+
+    /**
+     * Marks the read as being a supplementary alignment.
+     *
+     * <p>Equivalent to the 0x800 <b>FLAG</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation calls the {@code addFlag(SAMFlag.SUPPLEMENTARY_ALIGNMENT, supplementaryAlignment)}.
+     *
+     * @param supplementaryAlignment {@code true} if the read is a supplementary alignment; {@code false} otherwise.
+     */
+    default void setSupplementaryAlignment(final boolean supplementaryAlignment) {
+        addFlag(SAMFlag.SUPPLEMENTARY_ALIGNMENT, supplementaryAlignment);
+    }
+
+
+    ///////////////////////////////////////////////////////////
+    // OPTIONAL FIELDS - typed getters/setters
+    ///////////////////////////////////////////////////////////
+
+    /**
+     * Gets the character value associated with the tag.
+     *
+     * <p>Equivalent to an attribute with <b>TYPE A</b> in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
+     *
+     * <p>Default implementation gets the attribute with {@link #getAttribute(String)}, checking if it is a {@link Character}.
+     *
+     * @param tag the attribute tag.
+     *
+     * @return the character value.
+     *
+     * @throws SAMException if the tag is invalid following the contract of {@link ReadAttribute#isValidTag(String)}.
+     * @throws SAMException if the attribute value is not a character.
+     */
+    default Optional<Character> getCharacterAttribute(final String tag) {
+        final Optional<Object> attr = getAttribute(tag);
+        if (!attr.isPresent()) {
+            return Optional.empty();
+        }
+        final Object val = attr.get();
+
+        if (val instanceof Character) {
+            return Optional.of((Character)val);
+        }
+        throw new SAMException("Value for tag " + tag + " is not a Character: " + val.getClass());
+    }
+
+    /**
+     * Gets the singed integer value associated with the tag.
+     *
+     * <p>Equivalent to an attribute with <b>TYPE i</b> or any <b>TYPE B</b> integer (signed or unsigned) in the <a href="http://samtools.github.io/hts-specs/SAMv1.pdf">SAM specifications</a>.
```

Comment:
what do you mean "or any type B integer"? thats an array...how will you put an array into an Integer?

Reply:
True, I don't know what I did here....

Reply:
Removed.